src/main/kotlin/lotto/LottoLine.kt

```diff
@@ -0,0 +1,26 @@
+package lotto
+
+class LottoLine(private val lottoBalls: List<LottoBall>) {
+    constructor(lottoInput: String) : this(lottoInput.split(",").map { LottoBall(it.trim().toInt()) })
+
+    init {
+        checkIsSixBalls()
+        checkIsUniqueBalls()
+    }
+
+    private fun checkIsSixBalls() {
+        require(lottoBalls.size == 6) { "로또 번호는 6개여야 합니다." }
+    }
+
+    private fun checkIsUniqueBalls() {
+        require(lottoBalls.toSet().size == 6) { "로또 번호는 중복되지 않아야 합니다." }
+    }
+
+    fun extractMatchCount(other: LottoLine): Int {
+        return lottoBalls.count { other.lottoBalls.contains(it) }
```

Comment:
other.lottoBalls.contains(it) 는 디미터 법칙을 지키지 못하는 코드로 보여요. 
```suggestion
        return lottoBalls.count { other.containsBall(it) }
```

객체에게 요청과 응답을 하는방식으로 캡슐화를 해보면 어떨까요

Reply:
같은 LottoLine 단에서의 비교라 아무 생각 없이 lottoBalls 안쪽을 꺼내 비교했는데, 
각 객체에게 메시지를 전달한다는 입장에서 보니 확실히 어색하게 느껴지네요! 🚗 🚗